### PR TITLE
III-6131 Normalize saved searches

### PR DESCRIPTION
**Pull request by grubolsch**
### Fixed
It was possible to enter faulty saved queries, that the system could not read.
This code cleans them up.
It does NOT change anything in the database, it just makes sure the GET request returns correctly formatted queries.

Includes unit tests and a feature test.

---
Ticket: https://jira.uitdatabank.be/browse/III-6131

### CODE REVIEW

**Comment by grubolsch** [comment]
In draft because I still want to do some end to end test but feel free to remark already on the code structure.

**Comment by grubolsch** [comment]
Oké, I finished end to end tests and everything should be good now.

File: src/SavedSearches/Properties/QueryString.php [hunk 1]
```diff
@@ -31,4 +31,22 @@ public static function fromURLQueryString(string $queryString): QueryString
 
         return new QueryString($queryArray['q']);
     }
+
+    public function clean(): self
```
Comment (LucWollants) [inline src/SavedSearches/Properties/QueryString.php:35]: Maybe override the `toString` method and do the cleaning inside that method?

File: tests/SavedSearches/Properties/QueryStringTest.php [hunk 2]
```diff
@@ -31,4 +31,27 @@ public function it_can_get_a_query_string_from_a_url_query_string(): void
...
+    /**
+     * @test
+     * @dataProvider dataproviderBrokenQueries
+     */
+    public function it_cleans_broken_queries(string $brokenQuery, string $fixedQuery): void
+    {
+        $this->assertEquals($fixedQuery, (new QueryString($brokenQuery))->clean()->toString());
+    }
+
+    public function dataproviderBrokenQueries(): array
```
[the remark this answers is not in the document]
Comment (grubolsch) [reply tests/SavedSearches/Properties/QueryStringTest.php:45]: Added more examples, but there is a (disappointing? hopeful?) lack of problems in the prod database, almost all queries have the same 2 problems, the urlencoded + and too much slashes in the date range.

**Comment by grubolsch** [comment]
I changed the approach of this PR, after more research I saw that a full urldecode() was not needed because all encoded chars was just the "+". Now I just replace that +, making it no longer harmful to run the clean operation multiple times on the same query.
This allows me to always run the clean code in the constructor, making it much more elegant and robust.

**Review by LucWollants — APPROVED** [review]

**Pull request merged**